### ISSUE TITLE
Update to cyrus-sasl 2.1.28

### ISSUE THREAD

**Issue by cmb69** [winlibs/cyrus-sasl#3]
The update is available at https://github.com/winlibs/cyrus-sasl/tree/cmb/2.1.28. An untested build (PHP 8.4 staging) at https://github.com/winlibs/winlib-builder/actions/runs/11653667109.

@matsuo claims there are relevant security fixes. This needs to be investigated; possibly we may want to roll out updates for stable PHP versions as well. Although it is not clear whether this would need re-builds of OpenLDAP per se (since our sasl builds are dynamic anyway).

Also I'm somewhat reluctant to roll out 2.1.28 at all. Apparently, upstream doesn't care much about Windows; the [installation guide](https://www.cyrusimap.org/sasl/sasl/windows.html#install-windows) is grossly outdated, and conflicts with [the build instructions in the sources](https://github.com/cyrusimap/cyrus-sasl/blob/cyrus-sasl-2.1.28/win32/README.md). Furthermore, 2.1.28 is not necessarily [ready for OpenSSL 1.1](https://github.com/cyrusimap/cyrus-sasl/blob/cyrus-sasl-2.1.28/win32/openssl.props#L9). And it can't even be built on Windows due to [grossly conflicting typedefs](https://github.com/winlibs/cyrus-sasl/commit/a14b88767fd11fb87704a94c8c34393b36f80eae#diff-16a51509cf66579c13b56a618b35bef50deb072e97b49ba2111d5678db592c7f). All in all, it might still be better than 2.1.27.

**Comment by matsuo** [comment]
@cmb69 
FYI, Cyrus SASL 2.1.28 contains 2 security fixes.
https://www.cyrusimap.org/sasl/sasl/release-notes/2.1/index.html#new-in-2-1-28

I found related issues:
- https://github.com/cyrusimap/cyrus-sasl/issues/587
- https://github.com/cyrusimap/cyrus-sasl/issues/639

**Comment by Neustradamus** [comment]
Yes important and I hope a new Cyrus SASL version too because 2.1.28 (2022-02-18) is old.
